### PR TITLE
Hierarchies: Add a learning page for hierarchy definitions

### PR DESCRIPTION
**Pull request by grigasp**
Part of https://github.com/iTwin/presentation/issues/423

### CODE REVIEW

File: packages/hierarchies/learning/HierarchyProcessing.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+   2.1. In case the hierarchy level definition is a `CustomHierarchyNodeDefinition`, it's simply mapped to the custom node.
+
+   2.2. In case the hierarchy level definition is an `InstanceNodesQueryDefinition`, the nodes are fetched from the iModel and parsed either by definition's `parseNode` function, or the default parser.
+
+3. Nodes are passed through the labels formatter.
+4. Nodes are passed through the hierarchy definition's `preProcessNode` function, if one is defined.
+5. Nodes are passed through the hiding processor that hides based on their `hideIfNoChildren` and `hideInHierarchy` processing flags. This step may require loading children for some nodes, e.g. to find out if they have children or get the children to replace the processed node.
+6. Nodes are passed through the grouping processor.
+7. Nodes' children flag is determined, if necessary. Unless the node already knows if it has children, the children are loaded to determine if the parent has children.
```
Comment (Yato333) [inline packages/hierarchies/learning/HierarchyProcessing.md:16]: I think this could be said in a more concise manner. Something like this:
```suggestion
7. If nodes' children flag is undefined, it is determined by loading them.
```

File: packages/hierarchies/learning/HierarchyDefinitions.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    return undefined;
+  },
+};
+```
+
+<!-- END EXTRACTION -->
+
+## Custom post-processing
+
+The post-processing step happens after nodes' hiding, grouping and children determination steps, but before sorting. The step allows modifying the nodes and is especially useful when grouping nodes need to be modified in some way.
```
Comment (Yato333) [inline packages/hierarchies/learning/HierarchyDefinitions.md:168]: If I were you, I'd reword the second sentence to avoid repeating modifying/modified and explain why this is useful: 

```suggestion
The post-processing step happens after nodes' hiding, grouping and children determination steps, but before sorting. This step allows to modify the nodes and is especially useful for the grouping nodes because this is the only way to customize them before completing the hierarchy level definition.
```

File: packages/hierarchies/learning/HierarchyProcessing.md (hunk 1 above)
Comment (grigasp) [reply packages/hierarchies/learning/HierarchyProcessing.md:16]: I like the idea to shorted what I had, but I'd like to reword a bit. What do you think about the following?

_7. If nodes' children flag is undefined, it is determined by loading the children._

File: packages/hierarchies/learning/HierarchyDefinitions.md (hunk 2 above)
Comment (grigasp) [reply packages/hierarchies/learning/HierarchyDefinitions.md:168]: What do you think about the following?

_The post-processing step allows to modify the nodes in their final form - fully customized, grouped and with children determined. Also, since the step happens after grouping, it gives a chance to apply modifications to grouping nodes._

File: packages/hierarchies/learning/HierarchyProcessing.md (hunk 1 above)
Comment (Yato333) [reply packages/hierarchies/learning/HierarchyProcessing.md:16]: That's also ok

File: packages/hierarchies/learning/HierarchyDefinitions.md (hunk 2 above)
Comment (Yato333) [reply packages/hierarchies/learning/HierarchyDefinitions.md:168]: That's much better

Comment (grigasp) [reply packages/hierarchies/learning/HierarchyDefinitions.md]: pushed

File: packages/hierarchies/learning/HierarchyProcessing.md (hunk 1 above)
Comment (grigasp) [reply packages/hierarchies/learning/HierarchyProcessing.md]: pushed

**Review by Yato333 — APPROVED** [review]
